### PR TITLE
Remove unnecessary interrupt call

### PR DESCRIPTION
**Pull request by tajila**
Remove unnecessary interrupt call

Signed-off-by: Tobi Ajila <atobia@ca.ibm.com>

### CODE REVIEW

**Comment by tajila** [comment]
jenkins test sanity plinux jdk19

File: src/java.base/share/classes/java/lang/Thread.java [hunk 1]
```diff
@@ -1765,7 +1765,6 @@ public void interrupt() {
             }
         }
         interrupted = true;
-        interrupt0();  // inform VM of interrupt
```
Comment (JasonFengJ9) [inline src/java.base/share/classes/java/lang/Thread.java:1768]: Is this required to interrupt itself which is always allowed, also is required when `nioBlocker` is `null`?

Comment (tajila) [reply src/java.base/share/classes/java/lang/Thread.java:1768]: yes, ive fixed that

**Comment by tajila** [comment]
jenkins test sanity plinux jdk19

File: src/java.base/share/classes/java/lang/Thread.java (hunk 1 above)
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/Thread.java:1768]: This removes the `interrupt0()` call for current thread which is also in OpenJ9 thread version.
The difference between OpenJDK and OpenJ9 threads is the `interrupt0()` for current thread is not synchronized w/ `interruptLock`, is it possible that is the cause of https://github.com/eclipse-openj9/openj9/issues/15491?
In addition, OpenJDK sets `interrupted = true` which is used by `isInterrupted()`.

Comment (tajila) [reply src/java.base/share/classes/java/lang/Thread.java:1768]: yes thats my assumption. Although I dont think this implementation is correct yet.

File: src/java.base/share/classes/java/lang/Thread.java [hunk 2]
```diff
@@ -1750,22 +1750,19 @@ public final void stop() {
...
-                    interrupted = true;
-                    interrupt0();  // inform VM of interrupt
-                    b.interrupt(this);
-                    return;
-                }
-            }
-        }
-        interrupted = true;
-        interrupt0();  // inform VM of interrupt
+		checkAccess();
```
Comment (JasonFengJ9) [inline src/java.base/share/classes/java/lang/Thread.java:1753]: I think `checkAccess()` is only needed `this != currentThread()`.

File: src/java.base/share/classes/java/lang/Thread.java [hunk 3]
```diff
@@ -1750,22 +1750,21 @@ public final void stop() {
...
+
+		synchronized (interruptLock) {
+			interrupt0();  // inform VM of interrupt
+			interrupted = true;
+
+			// thread may be blocked in an I/O operation
+			Interruptible b = nioBlocker;
+			if (b != null) {
+				b.interrupt(this);
+				return;
```
Comment (JasonFengJ9) [inline src/java.base/share/classes/java/lang/Thread.java:1765]: nit: this return is redundant now.

Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/Thread.java:1765]: The PR should be valid - https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK19_ppc64le_linux/184/

Comment (tajila) [reply src/java.base/share/classes/java/lang/Thread.java:1765]: will change this once tests passes

**Comment by tajila** [comment]
@JasonFengJ9 changes are ready

**Pull request merged**